bootstrap/download-streamine-artifacts.py

```diff
@@ -0,0 +1,183 @@
+		content = urllib2.urlopen(component_bundles_url)
+		resp_json = json.load(content)
+		entities_maven_deps = filter(lambda x: 'mavenDeps' in x and x['mavenDeps'] != None, resp_json['entities'])
+		maven_deps_list.extend(map(lambda x: x['mavenDeps'], entities_maven_deps))
+	return maven_deps_list
+
+
+def write_downloaded_information_to_output_file(output_file, downloaded_artifacts):
+	with open(output_file, 'w') as fw:
+		for artifact_info, downloaded_location in downloaded_artifacts.iteritems():
+			fw.write("%s\t%s\n" % (artifact_info, downloaded_location))
+
+
+def main():
+	if len(sys.argv) < 3:
+		print("USAGE: [python] %s [streamline config file] [output file]" % sys.argv[0])
+		sys.exit(1)
+
+	config_file = sys.argv[1]
+	output_file = sys.argv[2]
+
+	print("> CONFIG FILE: %s" % config_file)
+	print("> OUTPUT FILE: %s" % output_file)
+
+	config_struct = load_config(config_file)
+
+	print("================ CONFIG ==================")
+	print("CATALOG ROOT URL: %s" % config_struct.catalog_root_url)
+	print("STORM HOME DIR: %s" % config_struct.storm_home_dir)
+	print("MAVEN REPOSITORY URL: %s" % config_struct.maven_repo_url)
+	print("HTTP PROXY URL: %s" % config_struct.http_proxy_url)
+	print("HTTP PROXY USERNAME: %s" % config_struct.http_proxy_username)
+	print("HTTP PROXY PASSWORD: %s" % config_struct.http_proxy_password)
+	print("STORM SUBMIT TOOL CLASSPATH: %s" % config_struct.get_storm_submit_tool_classpath())
+
+	component_bundles_kind = load_component_bundles_kind(config_struct)
+
+	maven_deps_list = collect_maven_deps_list(config_struct, component_bundles_kind)
+
+	print("============== COLLECTED Maven Dependencies ==============")
+	for maven_deps in maven_deps_list:
+		print(maven_deps)
+
+	downloaded_artifacts = {}
+
+	for maven_deps in maven_deps_list:
+		submit_tool_cmd = config_struct.build_storm_submit_tool_command(maven_deps)
+		print("============== Running COMMAND: %s" % submit_tool_cmd)
+		p = subprocess.Popen(submit_tool_cmd, stdout=subprocess.PIPE)
+		stdout, errors = p.communicate()
+		print("============== Return value of SUBMIT TOOL: %s" % p.returncode)
+		if p.returncode != 0:
+			raise RuntimeError("dependency handler returns non-zero code: code<%s> syserr<%s>" % (p.returncode, errors))
+		if not isinstance(stdout, str):
+			stdout = stdout.decode('utf-8')
+		ret_json = json.loads(stdout)
+		downloaded_artifacts.update(ret_json)
+	
+	print("============== Writing downloaded information to file: %s" % output_file)
+	write_downloaded_information_to_output_file(output_file, downloaded_artifacts)
+
+	print("DONE!")
+
+	print("Please follow below steps:")
+	print("1. archive 'local-repo' in current directory which downloaded artifacts are placed.")
+	print("2. send archived file to the machines which streamline is installed and don't have access to internet.")
+	print("3. extract file to 'local-repo' in root directory of streamline installation.")
```

Comment:
@HeartSaVioR , In some installations I have seen the local_repo getting created under /tmp (e.g. /tmp/hsperfdata_streamline/local-repo) . 

Does aether create/looks for the local-repo directory under the process CWD by default? 

If so we should
1. Create an empty "local-repo" directory under "base dir" as part of streamline distribution with "rwx" for others.
2. Change the CWD of streamline to "base dir" in the launch script (E.g. `(cd $base_dir && exec ....)` )
3. Now users can place the jars under local-repo so that aether picks it up correctly.

Reply:
@arunmahadevan 

> In some installations I have seen the local_repo getting created under /tmp (e.g. /tmp/hsperfdata_streamline/local-repo).
> Does aether create/looks for the local-repo directory under the process CWD by default?

Sounds weird. Not sure why it is created under such directory, because it should use CWD of where it launches storm-submit-tools.

```
            String localMavenRepoPath = getOrDefaultLocalMavenRepositoryPath("local-repo");

            // create root directory if not exist
            Files.createDirectories(new File(localMavenRepoPath).toPath());
```

Note that we should consider which account runs `storm-submit-tool`, and which CWD is it. Since it is executed as a part of `storm submit`, it is affected how `storm` and `storm.py` modifies its CWD, so the location might be probably what we don't expect.

I'm avoiding to fix Storm stuff, but looks like I need to fix storm-submit-tool as well as storm.py script to receive custom local repository directory, so the local repository directory is decided as such priorities:

1. custom local repository directory if given
2. `.m2/repository` in account home directory if exists
3. CWD/local-repo

which requires both fixes of Storm and Streamline. This script can be updated to receive custom local repository directory afterwards, which users should know where is it.